app/src/main/java/cmpe277/project/skibuddy/UserProfileActivity.java

```diff
@@ -0,0 +1,91 @@
+package cmpe277.project.skibuddy;
+
+import android.app.Activity;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.util.Log;
+import android.widget.ImageView;
+import android.widget.TextView;
+import java.io.InputStream;
+import java.util.ArrayList;
+
+public class UserProfileActivity extends Activity {
+    TextView name;
+    TextView tagline;
+    ImageView imageView;
+
+    ArrayList<String> arrStr = new ArrayList<String>();
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.user_profile);
+
+        arrStr = getIntent().getStringArrayListExtra("UserProfileActivity");
```

Comment:
The one you're creating doesn't have it, but the one @AkankshaNagpal creates will only hold the UUID. Use the key from BundleKeys to get it. 

         Bundle bundle = getIntent().getExtras();
         if (bundle != null) {
            String userUUID = bundle.getString(BundleKeys.UUID_KEY);
            s.getUser(UUID.fromString(userUUID), new ServerCallback 

etc.